core/src/services/gcs/writer.rs

```diff
@@ -32,11 +32,28 @@ pub struct GcsWriter {
 
     op: OpWrite,
     path: String,
+    upload_location: Option<String>,
+    already_uploaded_chunk: u64,
```

Comment:
`chunk` is not a native idea in gcs. How about use `written` to represents bytes that already written into gcs?